numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1549,6 +1549,61 @@ gentype_@name@(PyObject *self, PyObject *args, PyObject *kwds)
 }
 /**end repeat**/
 
+static PyObject *
+gentype_dunder_round(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    static char *kwlist[] = {"ndigits", NULL};
+    PyObject *ndigits = NULL;
```

Comment:
Why not initialize this to `Py_None`? Then you can remove the `NULL` checks below.

Reply:
Will that cause a `Py_DECREF` inside `PyArg_ParseTupleAndKeywords`?

Reply:
No, `PyArg_ParseTupleAndKeywords` does not look at the values of the passed pointers, it only sets them.